src/pages/Entries/Entries.tsx

```diff
@@ -15,26 +18,72 @@ export default function Entries() {
   const [editing, setEditing] = useState(false);
   const prevSelectedEntryId = useRef<number | undefined>(undefined);
   const { user } = useUser();
+  const { api } = useApi();
+  const queryClient = useQueryClient();
+  const numStillMutating = useIsMutating({ mutationKey: MutationKeys.MUTATE_HAPPINESS });
+  const [networkingState, setNetworkingState] = useState(
+    Constants.LOADING_MUTATION_TEXT.toString(),
+  );
+
+  const updateHappinessTimeout = useRef<number | undefined>(undefined);
+  const updateHappiness = () => {
+    //@ts-ignore TODO get rid of user_id field once backend is actually updated
+    const { id, author, user_id, ...happinessToPost } = selectedEntry;
+    updateEntryMutation.mutate(happinessToPost);
+  };
 
   useEffect(() => {
-    if (
-      selectedEntry &&
-      prevSelectedEntryId &&
-      prevSelectedEntryId.current !== selectedEntry?.id
-    ) {
-      setEditing(false);
-    }
     if (selectedEntry) {
-      prevSelectedEntryId.current = selectedEntry.id;
+      if (selectedEntry.value === -1) {
+        setNetworkingState(Constants.NO_HAPPINESS_NUMBER);
+      } else {
+        setNetworkingState(Constants.LOADING_MUTATION_TEXT);
+        prevSelectedEntryId.current = selectedEntry.id;
+        clearTimeout(updateHappinessTimeout.current);
+        updateHappinessTimeout.current = setTimeout(updateHappiness, 1000);
+      }
     }
   }, [selectedEntry]);
 
+  // General mutation function for updating Happiness entries
+  const updateEntryMutation = useMutation({
+    mutationFn: (newHappiness: HappinessPost) => {
+      return api.post<Happiness>("/happiness/", newHappiness);
+    },
+    mutationKey: MutationKeys.MUTATE_HAPPINESS,
+  });
+
+  const deleteHappinessMutation = useMutation({
+    mutationFn: () =>
+      api.delete(`/happiness/?id=${selectedEntry?.id}`),
+    mutationKey: MutationKeys.MUTATE_HAPPINESS
+  });
+
+  // Update the networking state displayed to the user based on updateEntryMutation result
+  useEffect(() => {
+    if (!selectedEntry || selectedEntry.value === -1) {
+      setNetworkingState(Constants.NO_HAPPINESS_NUMBER);
+      return;
+    }
+    if (numStillMutating > 0) {
+      setNetworkingState(Constants.LOADING_MUTATION_TEXT);
+      return;
+    }
+    if (updateEntryMutation.isError || deleteHappinessMutation.isError) {
+      setNetworkingState(Constants.ERROR_MUTATION_TEXT);
+      return;
+    }
+    queryClient.invalidateQueries(QueryKeys.FETCH_HAPPINESS);
```

Comment:
so is the intention of this to invalidate the old fetch so the scrollable calendar/sidebar is updated?

Reply:
yes, this line basically tells react query, all data that depends on happiness is now out of date, so it knows that the components that use `useQuery` hook with the query key for fetching happiness must be refetched.